plugins/analysis-icu/src/main/java/org/elasticsearch/index/analysis/IcuCollationTokenFilterFactory.java

```diff
@@ -55,7 +56,7 @@ public IcuCollationTokenFilterFactory(IndexSettings indexSettings, Environment e
             Exception failureToResolve = null;
             try {
                 rules = Streams.copyToString(Files.newBufferedReader(environment.configFile().resolve(rules), Charset.forName("UTF-8")));
-            } catch (IOException | SecurityException e) {
+            } catch (IOException | SecurityException | InvalidPathException e) {
```

Comment:
Can you explain in a bit more detail where the InvalidPathException that you catch here is coming from? The test that is added in this PR passes for me (I get a NoSuchFileException, which is caught as a IOException) even if I revert this change. It would be good to have a test that fails if this change isn't present.

Reply:
Ok, strange as I get the InvalidPathException from the FilterPath.resolve(). However, instead of checking our environments you can just try to create an index using the rules:

PUT myindex
{
	"settings" : {
		"analysis": {
			"analyzer": {
				"collation": {
					"tokenizer": "keyword",
					"type": "custom",
					"filter": [ "mycollator" ]
				}
			},
			"filter": {
				"mycollator": {
					"type": "icu_collation",
					"rules": "& a < b"
				}
			}
		}
	}
}

Results in (tested on 5.5.2):
{
    "error": {
        "root_cause": [
            {
                "type": "invalid_path_exception",
                "reason": "Illegal char <<> at index 4: & a < b"
            }
        ],
        "type": "invalid_path_exception",
        "reason": "Illegal char <<> at index 4: & a < b"
    },
    "status": 400
}

Reply:
Strangely this doesn't reproduce for me either on 5.5.2. I can use the collation rule "&a < b" (I removed one whitespace here, but even with it I don't get the error) and the change takes effect as expected.
Maybe we should dig into the cause for this issue first, would it be possible for you to open an issue and provide some more feedback about your system etc... so we can try to reproduce and see if this is a bug?

Reply:
Created an issue https://github.com/elastic/elasticsearch/issues/27208

Reply:
Thanks, I'll add my observations there, it look like you are using Windows, maybe the error is tied to that system. Lets see if there is somebody else who can reproduce this on that system.